### PR TITLE
rpc: add context toggle to getProgramAccounts

### PR DESCRIPTION
**Pull request by 00nktk**
#### Problem
There is no way in current implementation to get `getProgramAccounts` response with context. 

#### Summary of Changes
Added an optional flag (`with_context`) in `getProgramAccounts` config to set the format of the response. The implementation is simple: if it is present and is `true`, the response will be wrapped in RpcResponse, otherwise it defaults to array. While it feels like returning RpcResponse should be the default behavior, there is a big chance that someone relies on the current format of the API. This might seem like not the most elegant solution, but it's backwards compatible.

### CODE REVIEW

**Review by mvines — COMMENTED** [review]
This is a nice improvement, thank you!

File: cli/src/cluster_query.rs [hunk 1]
```diff
@@ -1681,6 +1681,7 @@ pub fn process_show_stakes(
             encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
             ..RpcAccountInfoConfig::default()
         },
+        with_context: None,
```
[the remark this answers is not in the document]
Comment (00nktk) [reply cli/src/cluster_query.rs]: Sure. Fixed

**Comment by CriesofCarrots** [comment]
Nice, I bet there are a few other rpc methods that could take advantage of `OptionalContext`

File: core/src/rpc.rs [hunk 2]
```diff
@@ -350,7 +361,8 @@ impl JsonRpcRequestProcessor {
         program_id: &Pubkey,
         config: Option<RpcAccountInfoConfig>,
         filters: Vec<RpcFilterType>,
-    ) -> Result<Vec<RpcKeyedAccount>> {
+        with_context: bool, // TODO: consider RpcProgramsConfig instead of separate arg
```
Comment (mvines) [inline core/src/rpc.rs:364]:
```suggestion
        with_context: bool,
```

File: docs/src/developing/clients/jsonrpc-api.md [hunk 3]
```diff
@@ -1780,7 +1780,7 @@ Returns all accounts owned by the provided program Pubkey
     "jsonParsed" encoding attempts to use program-specific state parsers to return more human-readable and explicit account state data. If "jsonParsed" is requested but a parser cannot be found, the field falls back to "base64" encoding, detectable when the `data` field is type `<string>`.
   - (optional) `dataSlice: <object>` - limit the returned account data using the provided `offset: <usize>` and `length: <usize>` fields; only available for "base58", "base64" or "base64+zstd" encodings.
   - (optional) `filters: <array>` - filter results using various [filter objects](jsonrpc-api.md#filters); account must meet all filter criteria to be included in results
-
+  - (optional) `withContext: bool` - wrap the returned array into RpcResponse JSON object.
```
Comment (mvines) [inline docs/src/developing/clients/jsonrpc-api.md:1783]:
```suggestion
  - (optional) `withContext: bool` - wrap the result in a RpcResponse JSON object.
```

**Review by mvines — APPROVED** [review]
lgtm with my documentation nits picked

**Comment by 00nktk** [comment]
> Nice, I bet there are a few other rpc methods that could take advantage of `OptionalContext`

Thanks! Glad to help!

**Comment by mvines** [comment]
Downstream projects build fix landing via https://github.com/solana-labs/solana-program-library/pull/1766/files, once it does I'll retrigger the failed build here

**Pull request merged**